File: backend/backend/src/main/java/com/example/backend/entity/Report.java
package com.example.backend.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "report")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Report {
    @Id
    @Column(name = "id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "report_form")
    private String reportForm;

    @Column(name = "fullname")
    private String fullName;

    @Column(name = "email")
    private String email;

    @Column(name = "phone_number")
    private String phoneNumber;

    @Column(name = "content")
    private String content;

    @Column(name = "state")
    private String state;

    @Column(name = "device_id")
    private String deviceId;

    @Column(name = "image_1")
    private String image1;

    @Column(name = "image_2")
    private String image2;

    @Column(name = "solving")
    private String solving;

    @Column(name = "ads_panel")
    private Integer adsPanel;

    @Column(name = "ads_position")
    private Integer adsPosition;
}

File: frontend/src/types/report.ts
import { AdsPanel, AdsPosition } from './ads';
import { ReportStatus } from './enum';

export type Report = {
  reportForm: ReportForm;
  adsPanel: AdsPanel;
  adsPosition: AdsPosition;
  report: AdsReport;
};

export type AdsReport = {
  id: number;
  fullName: string;
  email: string;
  phoneNumber: string;
  content: string;
  state: ReportStatus;
  deviceId: string;
  image1: string;
  image2: string;
  solving: string;
  reportForm: string;
  adsPanel: number;
  adsPosition: 3;
};

export type ReportForm = {
  title: string;
  color: string;
  icon: string;
};
